ci/supervisor/config/config.go

```diff
@@ -0,0 +1,329 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package config contains functions related to config files.
+package config
+
+import (
+	"fmt"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// yamlMap is a generic type for YAML files
+type yamlMap = map[interface{}]interface{}
+
+// EnvVar holds a environment variable (name and value)
+type EnvVar struct {
+	Name, Value string
```

Comment:
I think this would work, but isn't it better to explicitly encode their yaml keys?